### ISSUE TITLE
When VLANs are used the correct network adapter is not selected

### ISSUE THREAD

**Issue by lightbord** [docsteer/sacnview#192]
On MacOS with multiple VLANs setup on my machine, I set the interface in settings to a VLAN interface, then the program restarts, then comes back up and it has seemingly chosen either a random VLAN on that adapter or the parent interface itself. It won't stay locked to a specific VLAN interface.

**Comment by marcusbirkin** [comment]
Ah well, that's an intersting one.
It stores the "which interface should I use" option as the MAC/hardware address of the interface. But of cause on a desktop machine using VLANs, each vlan interface will have the same MAC address. 

We could also store the interface name, and if the appliction detects multiple interfaces with the same MAC address, then combine that with the saved interface name.

Could you run 'ipconfig' from the terminal and post the output please, this will allow us see if the interface names are shared as well.

**Comment by lightbord** [comment]
MacOS does use different interface names for each VLAN (ex vlan0, vlan1, vlan2, etcetera). VLAN interfaces share mac addresses. I was looking at the saved settings file a little while ago and noticed it saved interfaces by MAC address too. I would agree this is likely the issue.

Here is the output of ifconfig.
```
vlan0: flags=8843<UP,BROADCAST,RUNNING,SIMPLEX,MULTICAST> mtu 1500
	ether 00:12:34:56:78:90 
	inet6 fe80::7288:6bff:fe80:7890%vlan0 prefixlen 64 scopeid 0xc 
	inet 10.101.44.1 netmask 0xffff0000 broadcast 10.101.255.255
	nd6 options=1<PERFORMNUD>
	vlan: 110 parent interface: en4
	media: autoselect (100baseTX <full-duplex>)
	status: active
vlan1: flags=8843<UP,BROADCAST,RUNNING,SIMPLEX,MULTICAST> mtu 1500
	ether 00:12:34:56:78:90 
	inet6 fe80::7288:6bff:fe80:7890%vlan1 prefixlen 64 scopeid 0xd 
	inet 10.140.44.1 netmask 0xffff0000 broadcast 10.140.255.255
	nd6 options=1<PERFORMNUD>
	vlan: 120 parent interface: en4
	media: autoselect (100baseTX <full-duplex>)
	status: active
```

**Comment by marcusbirkin** [comment]
Try the auto build of #8c24473 https://file.io/D362aW

**Comment by lightbord** [comment]
The new build has fixed the issue.

**Comment by lightbord** [comment]
@marcusbirkin pardon the question I am not experienced with auto-build systems but am curious! How do you get links like the one you posted up there to the current build? I saw there was a branch for this issue that built on travis, how do you get that https://github.com/docsteer/sacnview/files/3394331/sACNView_2.1.0-beta.4-7-g8c24473.zip link?

**Comment by marcusbirkin** [comment]
There are two continuous integration services linked to the repo.
Every time a commit is pushed they spin up virtual machines and produce builds on different platforms.
So while I generally program with a Linux OS (and Tom I believe is MacOS based) we can, within minutes, see if the code also compiles on Windows, MacOS, and Linux.
I simply download the MacOS build artifact from the logs and attached it to the thread, having only tested if the solution would work on Linux.

**Comment by docsteer** [comment]
FWIW, this now applies on PCs also. Windows 10 allows you to create VLANs on a single NIC, but the combination name-and-MAC should solve it there also.

**Comment by docsteer** [comment]
Fixed in v2.1.1
